hudi-common/src/main/java/org/apache/hudi/common/config/LockConfiguration.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.config;
+
+import java.util.Properties;
+
+/**
+ * Configuration for managing locks. Since this configuration needs to be shared with HiveMetaStore based lock,
+ * which is in a different package than other lock providers, we use this as a data transfer object in hoodie-common
+ */
+public class LockConfiguration {
```

Comment:
whats the reason to have this in `hudi-common`? locking is only used by writing correct.

Reply:
So we need to pass the same configs to `HiveMetastoreBasedLockProvider` which is in hudi-hive-sync as well as `ZookeeperBasedLockProvider` which is in hudi-client. This object helps to keep the config reading from the same class.